app/src/vanilla/java/cm/aptoide/pt/search/view/SearchResultPresenter.java

```diff
@@ -364,4 +390,124 @@ private void doFirstSearch() {
+              view.setSuggestionsList(queryResults);
+              view.toggleSuggestionsView();
+            }))
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(__ -> {
+        }, e -> crashReport.log(e));
+  }
+
+  private void handleQueryTextSubmitted() {
+    view.getLifecycle()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(__ -> view.searchSetup())
+        .flatMap(__ -> getDebouncedQueryChanges())
+        .filter(data -> data.hasQuery() && data.isSubmitted())
+        .observeOn(viewScheduler)
+        .doOnNext(data -> {
+          view.collapseSearchBar(false);
+          view.hideSuggestionsViews();
+          navigator.navigate(data.getQuery());
+          if (data.isSuggestion()) {
+            analytics.searchFromSuggestion(data.getQuery(), data.getPosition());
+          } else {
+            analytics.search(data.getQuery());
+          }
+        })
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(__ -> {
+        }, e -> crashReport.log(e));
+  }
+
+  private void handleSuggestionClicked() {
+    view.getLifecycle()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(__ -> view.listenToSuggestionClick())
+        .filter(data -> data.hasQuery() && data.isSubmitted())
+        .observeOn(viewScheduler)
+        .doOnNext(data -> {
+          view.collapseSearchBar(false);
+          view.hideSuggestionsViews();
+          navigator.navigate(data.getQuery());
+        })
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(__ -> {
+        }, e -> crashReport.log(e));
+  }
+
+  private void handleToolbarClick() {
+    view.getLifecycle()
+        .filter(event -> event == View.LifecycleEvent.CREATE)
+        .flatMap(__ -> view.toolbarClick())
+        .doOnNext(__ -> analytics.searchStart(SearchSource.SEARCH_TOOLBAR, true))
+        .doOnNext(__ -> view.focusInSearchBar())
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(__ -> {
+        }, err -> crashReport.log(err));
+  }
+
+  private void handleSearchMenuItemClick() {
+    view.getLifecycle()
+        .filter(event -> event == View.LifecycleEvent.RESUME)
+        .flatMap(__ -> view.searchMenuItemClick())
+        .doOnNext(__ -> analytics.searchStart(SearchSource.SEARCH_ICON, true))
+        .doOnNext(__ -> view.focusInSearchBar())
+        .compose(view.bindUntilEvent(View.LifecycleEvent.PAUSE))
+        .subscribe(__ -> {
+        }, err -> crashReport.log(err));
+  }
+
+  private void handleClickOnBottomNav() {
+    view.getLifecycle()
+        .filter(lifecycleEvent -> lifecycleEvent.equals(View.LifecycleEvent.CREATE))
+        .flatMap(created -> bottomNavigator.navigationEvent())
+        .filter(navigated -> view.hasResults())
+        .doOnNext(__ -> view.scrollToTop())
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
```

Comment:
ObserveOn(mainThread). After 171 goes to dev filter for only Search events using ObjectMapper that it's the 171 ticket

Reply:
Fixed